tobac/segmentation/watershed_segmentation.py

```diff
@@ -1225,33 +1230,53 @@ def segmentation(
         in coords.
     """
     import pandas as pd
-    from iris.cube import CubeList
 
-    logging.info("Start watershedding 3D")
+    time_var_name: str = "time"
+    seg_out_type: str = "int64"
+
+    logging.info("Start watershedding")
 
     # check input for right dimensions:
     if not (field.ndim == 3 or field.ndim == 4):
         raise ValueError(
             "input to segmentation step must be 3D or 4D including a time dimension"
         )
-    if "time" not in [coord.name() for coord in field.coords()]:
+    try:
+        ndim_time = internal_utils.find_axis_from_coord(field, time_var_name)
+    except ValueError as exc:
         raise ValueError(
-            "input to segmentation step must include a dimension named 'time'"
-        )
+            "input to segmentation step must include a dimension named '{0}'".format(
+                time_var_name
+            )
+        ) from exc
 
-    # CubeList and list to store individual segmentation masks and feature DataFrames with information about segmentation
-    segmentation_out_list = CubeList()
+    # create our output dataarray
+    segmentation_out_data = (
+        field.astype("int64", casting="unsafe")
+        .rename("segmentation_mask")
+        .assign_attrs(threshold=threshold)
+    )
     features_out_list = []
 
-    # loop over individual input timesteps for segmentation:
-    # OR do segmentation on single timestep
-    field_time = field.slices_over("time")
+    # Iris workaround: convert cftime to datetime64
 
-    for i, field_i in enumerate(field_time):
-        time_i = field_i.coord("time").units.num2date(field_i.coord("time").points[0])
-        features_i = features.loc[features["time"] == np.datetime64(time_i)]
+    if np.issubdtype(features["time"].dtype, np.datetime64):
+        # we are (likely) a numpy datetime
+        all_times = features["time"]
+    else:
+        all_times = features["time"].map(np.datetime64)
+
+    for i_time, time_i in enumerate(field.coords[time_var_name]):
```

Comment:
We should think of more clearly differentiated variable names than `i_time` and `time_i`, but I can't think of any good suggestions right now